### PR TITLE
LOK-2748: FE: Redesign Service Discovery UI

### PR DESCRIPTION
**Pull request by faizanjanjua**
## Description

Part 1:
Create Typescript interface definitions for the different kinds of Service Discovery config fields. In the BFF, these are in a Json string, but they should have a certain known “shape”.

We’ll need interfaces for HttpConfig, HttpsConfig, NtpConfig, eventually also for WS-Man configurations. These should probably all derive from a ServiceConfig interface, even if it’s empty (a marker interface).

Then need to map the Json string to and from the appropriate interface type when receiving from or sending to the BFF. Internally, we use the objects with config being one of these interfaces.

For now, can just do this for the marker interface and HTTP/HTTPS interfaces.

Part 2: 
Redesign Service Discovery UI
Validate Service Discovery
Save Service Discovery

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2748

### CODE REVIEW

File: ui/src/dtos/discovery.dto.ts [hunk 1]
```diff
@@ -230,7 +274,7 @@ export const clientToServerValidation = async (
   let isValid = true
   let validationErrors = {}
   const convertedDiscovery = discoveryFromClientToServer(selectedDiscovery)
-
+  console.log(convertedDiscovery , 'converted Discovery')
```
Comment (Shahbaz-dataq) [inline ui/src/dtos/discovery.dto.ts:279]: Remove log

File: ui/src/components/Discovery/ServiceDiscoveryModal.vue [hunk 2]
```diff
@@ -0,0 +1,130 @@
+<template>
```
Comment (Shahbaz-dataq) [inline ui/src/components/Discovery/ServiceDiscoveryModal.vue:1]: Rename file to `DiscoveryTargetModal.vue`

File: ui/src/components/Discovery/ServiceDiscoveryModal.vue [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+});
+
+const onSelectService = (service: any) => {
+  const serviceSelected = discoveryStore.SelectedDiscoveryServices
+  serviceSelected.label = service?.label
+  serviceSelected.type = service?.type
+  serviceSelected.port = service?.port
+};
+
+// onMounted(() => {
```
Comment (Shahbaz-dataq) [inline ui/src/components/Discovery/ServiceDiscoveryModal.vue]: remove commented code

File: ui/src/types/discovery.d.ts [hunk 4]
```diff
@@ -155,3 +159,31 @@ export interface Service {
...
+
+export interface HttpConfig extends ServiceConfig {}
+export interface HttpsConfig extends ServiceConfig {}
+
+export type ConfigType = HttpConfig | HttpsConfig;
+export interface ConfigItem {
+  type: "HTTP" | "HTTPS";
+  config: ConfigType | string;
+}
+export interface ServiceConfigCollection {
```
Comment (Shahbaz-dataq) [inline ui/src/types/discovery.d.ts:187]: remove unused interface `ServiceConfigCollection`

File: ui/src/types/discovery.d.ts [hunk 5]
```diff
@@ -112,6 +112,10 @@ export interface DiscoveryStore {
   validationErrors: DiscoveryStoreErrors;
   disableSave: boolean;
   validateOnKeyUp: boolean;
+  SelectedDiscoveryServices : newServiceDiscoveryInModal;
+  IsServiceCreationModalVisible : boolean;
+  ServicesAndPortNumbers : Array<any>;
```
Comment (Shahbaz-dataq) [inline ui/src/types/discovery.d.ts:117]: Remove `ServicesAndPortNumbers`

File: ui/src/types/discovery.d.ts [hunk 6]
```diff
@@ -112,6 +112,10 @@ export interface DiscoveryStore {
   validationErrors: DiscoveryStoreErrors;
   disableSave: boolean;
   validateOnKeyUp: boolean;
+  SelectedDiscoveryServices : newServiceDiscoveryInModal;
```
Comment (Shahbaz-dataq) [inline ui/src/types/discovery.d.ts]: Rename `SelectedDiscoveryServices` to `selectedServiceDiscoverytarget`

File: ui/src/types/discovery.d.ts [hunk 7]
```diff
@@ -155,3 +159,31 @@ export interface Service {
   type?: string;
   port?: number;
 }
+
+export interface newServiceDiscoveryInModal{
```
Comment (Shahbaz-dataq) [inline ui/src/types/discovery.d.ts]: Rename `newServiceDiscoveryInModal` to `newServiceDiscoveryTarget`

File: ui/src/types/discovery.d.ts [hunk 8]
```diff
@@ -155,3 +159,31 @@ export interface Service {
...
+  id? : number;
+  target?: string;
+  label?: string;
+  type?: string;
+  port?: number;
+  status? : string
```
Comment (Shahbaz-dataq) [inline ui/src/types/discovery.d.ts:169]: rename `status` to `responseCode`

File: ui/src/components/Discovery/ServiceDiscoveryModal.vue [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        <FeatherInput
+          v-model.trim="discoveryStore.SelectedDiscoveryServices.port"
+          label="Port"
+          v-focus
+        />
+      </div>
+      <div class="services-protocol">
+        <FeatherInput
+          v-model.trim="discoveryStore.SelectedDiscoveryServices.status"
+          label="Status"
```
Comment (Shahbaz-dataq) [inline ui/src/components/Discovery/ServiceDiscoveryModal.vue:55]: Changes label from `Status` to `responseCode`

**Review by Shahbaz-dataq — CHANGES_REQUESTED** [review]
Need these changes

**Review by synqotik — APPROVED** [review]
Overall LGTM. I have a few nitpicks, but they aren't that important. Let's get this merged so we can do the demo this week.

**Review by Shahbaz-dataq — APPROVED** [review]

**Pull request merged**